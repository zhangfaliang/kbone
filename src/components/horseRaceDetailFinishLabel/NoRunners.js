import React from 'react';
import styles from './index.less';

const NoRunners = ({ prefixCls = 'default', noRunText = '', noRunners = [''] }) => {
  return (
    <div className={styles[`${prefixCls}-no-runners`]}>
      <span className={styles.noRunText}>{noRunText}</span>
      {noRunners.map(noRunner => (
        <span key={noRunner} className={styles.noRunner}>
          {noRunner}
        </span>
      ))}
    </div>
  );
};

NoRunners.COMPONENT_NAME = 'NORUNNERS';

export default NoRunners;
