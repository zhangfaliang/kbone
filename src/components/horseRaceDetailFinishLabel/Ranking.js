import React from 'react';
import styles from './index.less';

const Ranking = ({ prefixCls = 'default', rankingText = '1st' }) => {
  return <span className={styles[`${prefixCls}-ranking`]}>{rankingText}</span>;
};

export default Ranking;
