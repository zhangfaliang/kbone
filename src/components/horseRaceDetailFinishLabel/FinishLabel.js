import React from 'react';
import { HorseImg, HorseName, HorseNum } from '../horseRaceInfo';
import Ranking from './Ranking';
import styles from './index.less';

const FinishLabel = ({
  prefixCls = 'finish',
  odds = '5.6',
  favText = '',
  rankNum = '',
  rankingText = '',
  picUrl,
  shortName = '',
  horseCode = '',
  noEntry = '',
}) => {
  return (
    <div className={styles[`${prefixCls}-label`]}>
      <div className={styles.left}>
        {rankingText && <Ranking prefixCls={prefixCls} rankingText={rankingText} />}
        {picUrl !== undefined && <HorseImg prefixCls={prefixCls} picUrl={picUrl} />}
        {HorseNum && <HorseNum prefixCls={prefixCls} horseCode={horseCode} />}
        {HorseName && <HorseName prefixCls={prefixCls} shortName={shortName} />}
      </div>
      {rankNum && <span className={styles.center}>{rankNum}</span>}
      <div className={styles.right}>
        {odds && <span>{odds} </span>}
        {favText && <span>{favText}</span>}
        {noEntry && <span className={styles.noEntry}>{noEntry} </span>}
      </div>
    </div>
  );
};

FinishLabel.COMPONENT_NAME = 'FINISHLABEL';

export default FinishLabel;
