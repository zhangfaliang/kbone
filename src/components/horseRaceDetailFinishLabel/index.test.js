import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import FinishLabel from './FinishLabel';
import NoRunners from './NoRunners';
import Ranking from './Ranking';

describe('测试 horseRaceDetailFinishLabel', async () => {
  it('FinishLabel', () => {
    expect.assertions(1);
    const wrapper = render(
      <FinishLabel rankingText="werr" picUrl="werwer" rankNum="1" favText="wer" noEntry="werwe" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  }),
    it('NoRunners', () => {
      expect.assertions(1);
      const wrapper = render(<NoRunners />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  it('Ranking', () => {
    expect.assertions(1);
    const wrapper = render(<Ranking />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
