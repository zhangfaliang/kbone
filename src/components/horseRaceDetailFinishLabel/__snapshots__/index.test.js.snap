// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 horseRaceDetailFinishLabel FinishLabel 1`] = `
<div
  class="finish-label"
>
  <div
    class="left"
  >
    <span
      class="finish-ranking"
    >
      werr
    </span>
    <img
      alt=""
      class="finish-horse-img"
      src="werwer"
    />
    <div
      class="finish-horse-num"
    />
    <div
      class="finish-horse-name"
    />
  </div>
  <span
    class="center"
  >
    1
  </span>
  <div
    class="right"
  >
    <span>
      5.6 
    </span>
    <span>
      wer
    </span>
    <span
      class="noEntry"
    >
      werwe 
    </span>
  </div>
</div>
`;

exports[`测试 horseRaceDetailFinishLabel NoRunners 1`] = `
<div
  class="default-no-runners"
>
  <span
    class="noRunText"
  />
  <span
    class="noRunner"
  />
</div>
`;

exports[`测试 horseRaceDetailFinishLabel Ranking 1`] = `
<span
  class="default-ranking"
>
  1st
</span>
`;
